kythe/docs/schema-overview.txt

```diff
@@ -0,0 +1,421 @@
+}
+------------------------------------------------------------------------
+
+`kcd.go`:
+
+[source,Go]
+----
+type Reader interface {
+  Revisions(_ context.Context, filter *RevisionsFilter, f func(Revision) error) error
+}
+----
+
+`memdb.go`:
+
+[source,Go]
+----
+type DB struct {
+...
+}
+...
+// Revisions implements a method of kcd.Reader.
+func (db DB) Revisions(_ context.Context, want *kcd.RevisionsFilter, f func(kcd.Revision) error) error {
+----
+
+Here we see the top level `Reader` interface again from `kcd.go`. The interface
+function `Revisions` from that same file has a `childof` edge pointing to the
+interface's node. The implementation of `Reader` that lives in `memdb.go` has a
+`satisfies` edge back to the parent interface, and its own `Revisions` function
+has an `overrides` edge pointing to the top level `Revisions` function from
+`kcd.go`. For C++ and Java, interfaces, abstract classes, and so on, have
+similar relationships in the Kythe schema.
+
+Note: For languages with explicitly-declared inheritance and interface
+implementation (such as Java), Kythe uses the `extends` edge label instead of
+`satisfies`.
+
+=== [[FeatureFunc]] Functions and Parameters
+
+Functions are represented by semantic nodes that have additional edges to
+indicate their relationship to the parameters of the function. These edges are
+named `param.0`, `param.1`, etc., where the ordinal reflects the order of
+declaration.
+
+As illustrated in this example, the parameters in turn may have their own
+declaration sites, following the conventions of the jump-to-definition example:
+
+[graphviz]
+------------------------------------------------------------------------
+digraph FuncyParam {
+  graph [fontname="helvetica"]
+  node  [fontname="helvetica";shape=box,fontsize=10]
+  edge  [fontname="helvetica",fontsize=10]
+
+  def [label="'Revisions'\nkcd.go";shape=ellipse]
+  def -> func [label="defines/binding"]
+  func [label="function Revisions"]
+  p0 [label="variable _"]
+  p1 [label="variable filter"]
+  p2 [label="variable f"]
+
+  // Note that the following 3 lines are only necessary to make sure the
+  // ordering of function params is correct, since graphviz likes to put f before
+  // filter.
+  {rank=same;p0 p1 p2}
+  p0 -> p1 [style=invis]
+  p1 -> p2 [style=invis]
+
+  func -> p0 [label="param.0"]
+  func -> p1 [label="param.1"]
+  func -> p2 [label="param.2"]
+  p0def [label="'_'\nkcd.go";shape=ellipse]
+  p1def [label="'filter'\nkcd.go";shape=ellipse]
+  p2def [label="'f'\nkcd.go";shape=ellipse]
+  p0def -> p0 [label="defines/binding"]
+  p1def -> p1 [label="defines/binding"]
+  p2def -> p2 [label="defines/binding"]
+}
+------------------------------------------------------------------------
+
+`kcd.go`:
+
+[source,Go]
+----
+type Reader interface {
+  Revisions(_ context.Context, filter *RevisionsFilter, f func(Revision) error) error
+----
+
+NOTE: In Go, the blank (`_`) parameter name is used when the syntax requires the
```

Comment:
We should just name the parameter and remove the note since it is not relevant.

Reply:
Sure, though it's no longer a snippet from the original source then.

But that probably doesn't matter too much, so done.